review/src/Weapon.java

```diff
@@ -0,0 +1,28 @@
+import java.util.UUID;
+abstract class Weapon implements Item {
+    UUID uuid = UUID.randomUUID();
+    String uuidAsString = uuid.toString();
+    String name;
+    int amount;
+    int amountLimit;
+    String type;
+    double power;
+
+    abstract void setType (String type);
+    abstract String getType();
+
+    abstract void setPower(double power);
+    abstract double getPower();
+
+    @Override public abstract void setName(String name);
```

Comment:
Implement all the methods from Item (setName, getName, setAmount, getAmount, setAmountLimit, getAmountLimit)